src/Backups/BackupsWorker.cpp

```diff
@@ -30,25 +30,79 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+namespace Stage = BackupCoordinationStage;
+
 namespace
 {
-    /// Coordination status meaning that a host finished its work.
-    constexpr const char * kCompletedCoordinationStatus = "completed";
+    std::shared_ptr<IBackupCoordination> makeBackupCoordination(const String & coordination_zk_path, const ContextPtr & context, bool is_internal_backup)
+    {
+        if (!coordination_zk_path.empty())
+        {
+            auto get_zookeeper = [global_context = context->getGlobalContext()] { return global_context->getZooKeeper(); };
```

Comment:
```suggestion
            auto get_zookeeper = [global_context = context->getGlobalContext()]
            {
                auto new_zookeeper = global_context->getZooKeeper();
                /// Avoid stale reads in case we reconnected to another replica
                new_zookeeper->sync("/");
                return new_zookeeper;
            };
```